<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1096</width>
    <height>905</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>401</width>
      <height>561</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="left_layout">
     <item>
      <layout class="QVBoxLayout" name="dro_layout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="3" column="1">
          <widget class="QLineEdit" name="z_wcs"/>
         </item>
         <item row="2" column="2">
          <widget class="QLineEdit" name="y_mcs"/>
         </item>
         <item row="3" column="2">
          <widget class="QLineEdit" name="z_mcs"/>
         </item>
         <item row="1" column="2">
          <widget class="QLineEdit" name="x_mcs"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Z</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Y</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QRadioButton" name="goto_button">
           <property name="text">
            <string>GOTO</string>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="y_wcs"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>X</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="x_wcs">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QRadioButton" name="set_button">
           <property name="text">
            <string>SET</string>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QRadioButton" name="show_button">
           <property name="text">
            <string>SHOW</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>WCS (mm)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Machine (mm)</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="mode_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>DISABLED</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="jog_layout">
       <item>
        <layout class="QGridLayout" name="jog_buttons">
         <item row="2" column="1">
          <widget class="QPushButton" name="yp_jog">
           <property name="text">
            <string>Y+</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QPushButton" name="xp_jog">
           <property name="text">
            <string>X+</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="xm_jog">
           <property name="text">
            <string>X-</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QPushButton" name="ym_jog">
           <property name="text">
            <string>Y-</string>
           </property>
          </widget>
         </item>
         <item row="4" column="3">
          <widget class="QPushButton" name="zm_jog">
           <property name="text">
            <string>Z-</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QPushButton" name="zp_jog">
           <property name="text">
            <string>Z+</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" name="jog_settings">
         <item row="1" column="2">
          <widget class="QLineEdit" name="jog_s_1">
           <property name="text">
            <string>250</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="jog_d_2">
           <property name="text">
            <string>1</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Jog Speed (mm/s)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="jog_d_1">
           <property name="text">
            <string>0.1</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="jog_set_1">
           <property name="text">
            <string/>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QRadioButton" name="jog_set_3">
           <property name="text">
            <string/>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLineEdit" name="jog_s_2">
           <property name="text">
            <string>500</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QRadioButton" name="jog_set_4">
           <property name="text">
            <string/>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Jog Distance (mm)</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QLineEdit" name="jog_s_4">
           <property name="text">
            <string>2000</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="jog_d_3">
           <property name="text">
            <string>10</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QRadioButton" name="jog_set_2">
           <property name="text">
            <string/>
           </property>
           <property name="autoExclusive">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLineEdit" name="jog_s_3">
           <property name="text">
            <string>1000</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="jog_d_4">
           <property name="text">
            <string>20</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="load_nc">
           <property name="text">
            <string>Load .nc</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="nc_preview">
           <property name="text">
            <string>Preview</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextBrowser" name="nc_browser"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>10</y>
      <width>511</width>
      <height>411</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="HOLDER">
     <item>
      <widget class="QWidget" name="placeholder" native="true"/>
     </item>
     <item>
      <layout class="QHBoxLayout" name="obj_inputs">
       <item>
        <widget class="QPushButton" name="load_obj">
         <property name="text">
          <string>Load .obj</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="clear_obj">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="reset_view">
         <property name="text">
          <string>Reset View</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>0</y>
      <width>151</width>
      <height>108</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QPushButton" name="feed_hold">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>60</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 0, 0);
color: rgb(255, 255, 255);</string>
       </property>
       <property name="text">
        <string>Feed Hold</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cycle_start">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>40</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(0, 255, 0);</string>
       </property>
       <property name="text">
        <string>Resume</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>430</y>
      <width>511</width>
      <height>74</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Send directly to GRBL</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="input_command"/>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="play_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>(Re)start .nc</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="reset_button">
         <property name="text">
          <string>Reset GRBL</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pause_button">
         <property name="text">
          <string>Pause .nc</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stop_button">
         <property name="text">
          <string>Stop .nc</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>570</y>
      <width>391</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="1">
      <widget class="QPushButton" name="speed_cp_button">
       <property name="text">
        <string>+10%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="feed_cp_button">
       <property name="text">
        <string>+10%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QPushButton" name="feed_reset_button">
       <property name="text">
        <string>-&gt;100%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="feed_label">
       <property name="text">
        <string>Feed:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QPushButton" name="speed_reset_button">
       <property name="text">
        <string>-&gt;100%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="feed_cm_button">
       <property name="text">
        <string>-10%</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="speed_cm_button">
       <property name="text">
        <string>-10%</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="speed_label">
       <property name="text">
        <string>Spindle:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>510</y>
      <width>511</width>
      <height>91</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Tools Involved</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextBrowser" name="tool_list"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1096</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
  <action name="actionOpen_nc">
   <property name="text">
    <string>Open .nc</string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>Connect to fluidnc</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
